core/services/pipeline/orm.go

```diff
@@ -333,11 +333,14 @@ func (o *orm) InsertFinishedRun(run *Run, saveSuccessfulTaskRuns bool, qopts ...
 }
 
 // DeleteRunsOlderThan deletes all pipeline_runs that have been finished for a certain threshold to free DB space
-func (o *orm) DeleteRunsOlderThan(ctx context.Context, threshold time.Duration) error {
-	// Added 1 minute timeout to account for big databases
-	q := o.q.WithOpts(pg.WithParentCtx(ctx), pg.WithLongQueryTimeout())
+//  This is guaranteed not to run longer than the timeout
+func (o *orm) DeleteRunsOlderThan(ctx context.Context, threshold time.Duration, timeout time.Duration) error {
+	start := time.Now()
+
+	q := o.q.WithOpts(pg.WithParentCtx(ctx), func(q *pg.Q) { q.QueryTimeout = timeout })
```

Comment:
Is there no `pg.WithTimeout` ? That might be handy if not.

Reply:
Looking at this again, I'd suggest not allowing a timeout at all and expecting the caller to set that on the passed context instead.

Reply:
Good idea, I'll try this